src/components/Auth/SignUpForm.js

```diff
@@ -4,26 +4,22 @@ import { Link, Redirect } from 'react-router-dom';
 import { Box, Button, TextField } from '@material-ui/core/';
 import axios from 'axios';
 import { useAuth } from './AuthContext';
+import { validationResolver, defaultValue } from './SignUpForm.schema';
+import { Form, Field } from '../form';
 
 const SignUpForm = ({ toggleActiveForm }) => {
-  const [firstName, setFirstName] = useState(null);
-  const [lastName, setLastName] = useState(null);
-  const [username, setUsername] = useState(null);
-  const [email, setEmail] = useState(null);
-  const [password, setPassword] = useState(null);
   const [errorMessage, setErrorMessage] = useState(null);
   const [isLoggedIn, setIsLoggedIn] = useState(false);
   const referer = '/profile';
   const auth = useAuth();
 
-  const handleSignup = e => {
-    e.preventDefault();
+  const onSubmit = formData => {
```

Comment:
we could deconstruct `username` `password` etc from `formData` here already.

Reply:
Good point -- e.g.

```
const onSubmit = ({username, password, firstName, lastName, email}) => {
```

Reply:
Done: https://github.com/codebuddies/frontend/pull/105/commits/f0196b0ddfb2c3af442e9598edae0fb9fc954500